### PR TITLE
mentat 0.1.17 (new formula)

### PR DESCRIPTION
**Pull request by jakethekoenig**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/m/mentat.rb [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+  desc "Mentat is the open source coding assistant that leverages GPT-4 to write code for you. Tell Mentat what you need from the command line and watch as it creates code directly in existing or new source files."
+  homepage "mentat.ai"
+  url "https://files.pythonhosted.org/packages/72/9d/1c68404a2944423bb6450934c1b948bf38179a35bb1045133b8a2b287531/mentat-0.1.17.tar.gz"
+  sha256 "cf46aa5f214bed825ce4d283010acd64ed4dd1f6ab3fa33f24cd9f6220cfb2b8"
+  license "Apache-2.0"
+  head "https://github.com/AbanteAI/mentat.git", branch: "main"
+
+  depends_on "python@3.11"
+  depends_on "rust" => :build # for tiktoken
```
Comment (p-linnane) [inline Formula/m/mentat.rb:12]:
```suggestion
  depends_on "rust" => :build # for tiktoken
  depends_on "pygments"
  depends_on "python-certifi"
  depends_on "python-packaging"
  depends_on "python-typing-extensions"
  depends_on "python@3.11"
  depends_on "six"
```

File: Formula/m/mentat.rb [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+  resource "openai" do
+    url "https://files.pythonhosted.org/packages/89/ee/84bbd0161090f0f24e8a2ac175e6b6a936289ee02e9d5da414ce14d3d332/openai-0.27.8.tar.gz"
+    sha256 "2483095c7db1eee274cebac79e315a986c4e55207bb4fa7b82d185b3a2ed9536"
+  end
+
+  resource "packaging" do
+    url "https://files.pythonhosted.org/packages/fb/2b/9b9c33ffed44ee921d0967086d653047286054117d584f1b1a7c22ceaf7b/packaging-23.2.tar.gz"
+    sha256 "048fb0e9405036518eaaf48a55953c750c11e1a1b68e0dd1a9d62ed0c092cfc5"
+  end
```
Comment (p-linnane) [inline Formula/m/mentat.rb:77]:
```suggestion

```

File: Formula/m/mentat.rb [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+  resource "requests" do
+    url "https://files.pythonhosted.org/packages/9d/be/10918a2eac4ae9f02f6cfe6414b7a155ccd8f7f9d4380d62fd5b955065c3/requests-2.31.0.tar.gz"
+    sha256 "942c5a758f98d790eaed1a29cb6eefc7ffb0d1cf7af05c3d2791656dbd6ad1e1"
+  end
+
+  resource "six" do
+    url "https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz"
+    sha256 "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926"
+  end
```
Comment (p-linnane) [inline Formula/m/mentat.rb:132]:
```suggestion

```

File: Formula/m/mentat.rb [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+  resource "tqdm" do
+    url "https://files.pythonhosted.org/packages/62/06/d5604a70d160f6a6ca5fd2ba25597c24abd5c5ca5f437263d177ac242308/tqdm-4.66.1.tar.gz"
+    sha256 "d88e651f9db8d8551a62556d3cff9e3034274ca5d66e93197cf2490e2dcb69c7"
+  end
+
+  resource "typing-extensions" do
+    url "https://files.pythonhosted.org/packages/3c/8b/0111dd7d6c1478bf83baa1cab85c686426c7a6274119aceb2bd9d35395ad/typing_extensions-4.7.1.tar.gz"
+    sha256 "b75ddc264f0ba5615db7ba217daeb99701ad295353c45f9e95963337ceeeffb2"
+  end
```
Comment (p-linnane) [inline Formula/m/mentat.rb:152]:
```suggestion

```

**Review by p-linnane — CHANGES_REQUESTED** [review]

**Comment by jakethekoenig** [comment]
@p-linnane thanks for the review! Is this [doc](https://docs.brew.sh/Python-for-Formula-Authors) out of date? It said I should generate python dependencies with `brew update-python-resources` but that's not the preferred way to specify them anymore? Also this [doc](https://docs.brew.sh/Formula-Cookbook) should maybe be updated to mention `brew style`.

**Comment by p-linnane** [comment]
It's accurate, we have just been vendoring more Python resources lately to save on build time. I attempted to push some changes to this PR that would allow us to skip those resources moving forward, but maintainers do not have the ability to push to your fork. You'll also want to squash the commits into one using the PR title as the message.

**Review by p-linnane — DISMISSED** [review]

**Pull request merged**